pkg/operator/controller/ingress/controller.go

```diff
@@ -134,9 +134,51 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(source.Kind(operatorCache, &configv1.Proxy{}), handler.EnqueueRequestsFromMapFunc(reconciler.ingressConfigToIngressController)); err != nil {
 		return nil, err
 	}
+	// Watch secrets in the operand namespace that are referenced as default certificates so that
+	// checkDefaultCertificate can determine whether changes to the default certificate affect upgradeability.
+	// This watch is NOT required for updating the router deployment, as its volume mount automatically
+	// handles updates to the default certificate.
+	if err := c.Watch(source.Kind(operatorCache, &corev1.Secret{}), handler.EnqueueRequestsFromMapFunc(reconciler.defaultCertificateSecretToIngressController), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 
+// defaultCertificateSecretToIngressController creates reconcile requests for
+// IngressControllers that reference the provided secret as their default certificate.
+func (r *reconciler) defaultCertificateSecretToIngressController(ctx context.Context, o client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+	secret := o.(*corev1.Secret)
+	controllers := &operatorv1.IngressControllerList{}
+	if err := r.cache.List(ctx, controllers, client.InNamespace(r.config.Namespace)); err != nil {
+		log.Error(err, "failed to list ingresscontrollers for default certificate secret", "related", o.GetSelfLink())
+		return requests
+	}
+	for _, ic := range controllers.Items {
+		if ic.Spec.DefaultCertificate != nil && ic.Spec.DefaultCertificate.Name == secret.Name {
```

Comment:
You might also consider using an index, à la the certificate-publisher controller's `secretToIngressController` function.

Reply:
I thought about it, but initially didn't add because it felt a bit unnecessary, but now that you pointed out that someone already wrote it, and I felt like I couldn't say no now. Added.

Reply:
Whoops, didn't realize you can't have two indexes as the same name.

I just updated the name, but do you think having two redundant indexes is a problem? I suppose I could share them, but that would create a dependency between the two control loops, which feels odd to me.

Reply:
> Whoops, didn't realize you can't have two indexes as the same name.
> 
> I just updated the name, but do you think having two redundant indexes is a problem? I suppose I could share them, but that would create a dependency between the two control loops, which feels odd to me.

It shouldn't cause any problems to have redundant indexes other than the cost of the redundancy in terms of code, memory, and computation.  However, I expect you could create the index in `pkg/operator/operator.go` and pass it to each controller that needed it.  

If you do that refactoring, please do it in a separate commit so that we can cherry-pick it to the main branch.  But maybe it's simplest, for this PR, just to live with the redundancy.

Reply:
Fair point. I lean towards leaving it simple. Unless you think duplicate index makes us worse off than just not using an index at all, I'm inclined to leave it as is. We are restricted to secrets in the `openshift-ingress` namespace, and my guess is that shouldn't see a whole lot of activity.